torch/csrc/jit/script/compiler.cpp

```diff
@@ -2774,6 +2773,23 @@ struct to_ir {
       Value* tuple_val,
       Value* idx_val) {
     auto tuple_typ = tuple_val->type()->cast<TupleType>();
+    auto elems = tuple_typ->elements();
+    if (!toIValue(idx_val) && elems.size() > 0) {
+      auto list = tryConvertToType(
+          loc,
+          *graph,
+          ListType::create(elems[0]),
+          tuple_val,
+          /*allow_conversions*/ true);
+      if (list == tuple_val) {
+        throw ErrorReport(loc)
+            << "Cannot index into a " << tuple_typ->python_str()
```

Comment:
This could be more clear, maybe something like `... non-constant index because it has multiple element types`

Reply:
Agreed that would be a little more clear, but then it wouldn't handle the zero-tuple case, and i think it's still pretty clear as is